### DISCUSSION TITLE
Default values for environment variables

### DISCUSSION THREAD

**Discussion by jayvdb** [sagiegurari/cargo-make#839] [Q&A] [answered]
Is there a way to set an environment variable, only if it is not already set?

i.e. I would like to have a value which defaults to `--all-features`, but would like for the user be able to set it to `-no-default-features`

**Comment by sagiegurari** [comment, marked as answer]
@jayvdb ya you can do something like the below:

```toml
PREFER_EXISTING = { value = "new", condition = { env_not_set = ["PREFER_EXISTING"] } }
```

see more at:

https://github.com/sagiegurari/cargo-make#conditional
https://github.com/sagiegurari/cargo-make#global-configuration
